day-01/y_part-1.cpp

```diff
@@ -0,0 +1,24 @@
+#include <algorithm>
+#include <iostream>
+#include <vector>
+
+int main() {
+
+  std::vector<int> left;
+  std::vector<int> right;
+
+  int l, r;
+  while (std::cin >> l >> r) {
+    left.push_back(l);
+    right.push_back(r);
+  }
+
+  std::sort(left.begin(), left.end());
+  std::sort(right.begin(), right.end());
+  int sum = 0;
+  // why do I need size_t here?
+  for (std::size_t i = 0; i < left.size(); i++)
+    sum += std::abs(left[i] - right[i]);
+
+  std::cout << "result: " << sum << std::endl;
```

Comment:
minor pet-peeve:

`std::endl` does two things: 
1. Insert a newline character into the stream.
2. Flush the stream.

While the first action is usually intended, the flush is unnecessary. The program will exit after this instruction, so any open buffers will be flushed anyway.